asserts/snap_asserts.go

```diff
@@ -473,6 +593,17 @@ func (snaprev *SnapRevision) checkConsistency(db RODatabase, acck *AccountKey) e
 	if err != nil {
 		return err
 	}
+	if otherProvenance {
+		decl := a.(*SnapDeclaration)
+		ras := decl.RevisionAuthority(snaprev.Provenance())
+		for _, ra := range ras {
+			if err := ra.Check(snaprev); err == nil {
+				return nil
+			}
+		}
+		return fmt.Errorf("snap-revision assertion with provenance %q for snap id %q is not signed by an authorized authority: %s", snaprev.Provenance(), snaprev.SnapID(), snaprev.AuthorityID())
```

Comment:
nitpick, but it can be risky to have a successful return here: if we add more code to the bottom of this function, we might forget about this early return and not execute it. I'd rather keep the early returns for the unhappy cases:
```suggestion
		foundSignature := false
		for _, ra := range ras {
			if err := ra.Check(snaprev); err == nil {
				foundSignature = true
				break
			}
		}
		if !foundSignature {
			return fmt.Errorf("snap-revision assertion with provenance %q for snap id %q is not signed by an authorized authority: %s", snaprev.Provenance(), snaprev.SnapID(), snaprev.AuthorityID())
		}
```